lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1156,16 +1156,33 @@ struct RocrQueue {
         STATUS_CHECK(status, __LINE__);
     }
 
-    RocrQueue(hsa_agent_t agent, size_t queue_size, HSAQueue *hccQueue)
+    RocrQueue(hsa_agent_t agent, size_t queue_size, HSAQueue *hccQueue, queue_priority priority)
     {
+        // Map queue_priority to hsa_amd_queue_priority_t
```

Comment:
It seems to me that it would be better that `_priority` has the type `queue_priority` rather than `hsa_amd_queue_priority_t` since the hsa priority is only needed in this ctor to set the priority by calling the hsa API.  This would save you from doing another reserve mapping when removing the queue from the pool

Reply:
The reverse mapping when removing the queue would only go away if we don't mind using enums values for array indexing. Otherwise there would still be a mapping. This can be removed in case we go with enum based array indexing.

Reply:
yes, switching to `queue_priority` for indexing is a precondition